pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -310,6 +303,24 @@ impl InMemoryLayer {
         self.oldest_pending_lsn
     }
 
+    /// Callback to notify when this layer is inserted to the global layer map.
+    /// From this point on, we need to keep the global layer map informed whenever
+    /// the memory consumption of this layer increases, by calling
+    /// [`update_layer_memory_stats`].
+    pub fn register_layer_id(&self, layer_id: LayerId) {
+        let mut inner = self.inner.write().unwrap();
+
+        inner.layer_id = Some(layer_id);
+    }
+
+    /// Callback from the global layer map, when this layer is removed from global
+    /// layer map.
+    pub fn unregister(&self) {
+        let mut inner = self.inner.write().unwrap();
+
+        inner.layer_id = None;
+    }
```

Comment:
I don't think this is bad, but just wondering will we ever remove a layer and then add it back to the global layer map?